### PR TITLE
[promtail] allow to inject templates

### PR DESCRIPTION
**Pull request by Wielewout**
In this PR the `extraVolumes` and `initContainers` can be filled in with templates in a parent chart.

For the `initContainers` it allows to for example inject a container that waits until kafka is ready. From the context of a parent chart, a service would be created where the name for example is `{{ .Release.Name }}-kafka`. This could for example be injected in `kcat` to check if kafka is ready.

It's also possible now to include a secret that's created in a parent chart (with the release name included in the name). This could be useful to inject a password using `password_file` with a file coming from a volume made with that secret.

This could for example look like the following in a values file of a parent chart.
Note that `{{ $.Release.Name }}-loki-gateway-auth` is a secret here with usernames as keys and the corresponding password as a value within it.

```yaml
promtail:
  config:
    enabled: true
    clients:
      - url: http://{{ .Release.Name }}-loki-gateway.{{ .Release.Namespace }}:80/loki/api/v1/push
        tenant_id: logs
        basic_auth:
          username: promtail
          password_file: /etc/promtail/secrets/password
  
  extraVolumeMounts:
    - name: loki-gateway-password
      mountPath: /etc/promtail/secrets

  extraVolumes:
    - name: loki-gateway-password
      secret:
        secretName: "{{ $.Release.Name }}-loki-gateway-auth"
        items:
          - key: promtail
            path: password
```

### CODE REVIEW

**Review by zanhsieh — APPROVED** [review]

**Comment by zanhsieh** [comment]
@zalegrala Just curious, who are the members of loki maintainers? This chart includes them in Chart.yaml but not win much of their attention.

**Comment by zalegrala** [comment]
I'm not sure @zanhsieh.  I think folks are encouraged to use the `loki` chart as mentioned here:  https://github.com/grafana/helm-charts/tree/main/charts/loki
The promtail functionality is probably better suited to be using the alloy chart, here:
https://github.com/grafana/alloy/tree/main/operations/helm/charts/alloy

I don't think there is a problem with the community maintaining extra charts here, but it would be good to get more community involvement, since don't have a test environment for much of this, so getting other community members to sanity check and approve what might break in their environment would be helpful.

**Comment by dimgogos** [comment]
Perhaps @camrossi, @Sheikh-Abubaker would be kind enough to have a look at this PR?

**Comment by Sheikh-Abubaker** [comment]
@Wielewout could you please explain the scope of this PR ?

File: charts/promtail/Chart.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name: promtail
 description: Promtail is an agent which ships the contents of local logs to a Loki instance
 type: application
 appVersion: 3.0.0
-version: 6.16.5
+version: 6.17.0
```
Comment (Sheikh-Abubaker) [inline charts/promtail/Chart.yaml:6]:
```suggestion
version: 6.16.6
```

I think patch version should be bumped.

File: charts/promtail/README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # promtail
 
-![Version: 6.16.5](https://img.shields.io/badge/Version-6.16.5-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: 3.0.0](https://img.shields.io/badge/AppVersion-3.0.0-informational?style=flat-square)
+![Version: 6.17.0](https://img.shields.io/badge/Version-6.17.0-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: 3.0.0](https://img.shields.io/badge/AppVersion-3.0.0-informational?style=flat-square)
```
Comment (Wielewout) [inline charts/promtail/README.md:3]:
```suggestion
![Version: 6.16.6](https://img.shields.io/badge/Version-6.16.6-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: 3.0.0](https://img.shields.io/badge/AppVersion-3.0.0-informational?style=flat-square)
```

Comment (Sheikh-Abubaker) [reply charts/promtail/README.md]: yeah you've to make changes here also!

**Review by Sheikh-Abubaker — APPROVED** [review]
LGTM! Thanks for the Contribution @Wielewout! keep them coming!!

**Pull request merged**